MOOSE fails to build against --enable-parmesh libMesh configurations

### Description of the enhancement or error report

I currently see the compilation error,

In file included from /h2/roystgnr/git/moose/framework/src/base/Moose.C:32:0:
/h2/roystgnr/git/moose/framework/include/mesh/PatternedMesh.h:54:15: error: ‘SerialMesh’ was not declared in this scope

### Rationale for the enhancement or information for reproducing the error

Building devel MOOSE against a --enable-parmesh libMesh build is probably enough to replicate; I can't be certain because I'm currently testing a libMesh branch with other alterations.

MOOSE is presumably relying on the default libMesh configuration, where "mesh.h" includes "serial_mesh.h"; including the required file specifically should make it more robust.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

No API or ABI changes should be necessary to fix; I'll put in a PR shortly.

--- a/framework/include/mesh/PatternedMesh.h
+++ b/framework/include/mesh/PatternedMesh.h
@@ -17,6 +17,8 @@
 
 #include "MooseMesh.h"
 
+#include "libmesh/serial_mesh.h"
+
 class PatternedMesh;
 
 template<>

